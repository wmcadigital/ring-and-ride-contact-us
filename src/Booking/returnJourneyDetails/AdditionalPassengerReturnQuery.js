import { useEffect } from "react";
import PropTypes from "prop-types";
import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import RadioGroup from "../../common/RadioGroup";
import Question from "../../common/Question";
import RadioButton from "../../common/RadioButton";
import FieldError from "../../common/FieldError";
import ProgressIndicator from "../../common/ProgressIndicator";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutReturnJourneySectionName from "./getAboutReturnJourneySectionName";

import { required } from "../../common/validation";

const AdditionalPassengerReturnQuery = ({ setAdditionalPassengerReturn }) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const bookingParty = formValues["bookingParty"];

  const error = stateApi.submitFailed
    ? stateApi.errors?.additionalPassengerReturn
    : null;

  const question =
    bookingParty === "mySelf"
      ? "Do you want to bring anyone else with you on the return journey?"
      : `Does ${formValues["firstName"]} want to bring anyone else with them on the return journey?`;

  useEffect(() => {
    if (formValues.additionalPassengerReturn) {
      setAdditionalPassengerReturn(formValues.additionalPassengerReturn);
    }
  }, [formValues.additionalPassengerReturn]);

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("3", stateApi)}
        sectionName={getAboutReturnJourneySectionName(stateApi)}
      />
      <Question text={question} />
      <RadioGroup error={error}>
        <label className="wmrards-fe-label" htmlFor="additionalPassengerReturn">
          For example, a carer or guardian. They do not need to be a registered
          Ring and Ride scheme user.
        </label>
        <FieldError text={error} />
        <RadioButton
          key={1}
          label="Yes"
          validation={required}
          value="yes"
          fieldName="additionalPassengerReturn"
        />
        <RadioButton
          key={2}
          label="No"
          validation={required}
          value="no"
          fieldName="additionalPassengerReturn"
        />
      </RadioGroup>
    </FormSection>
  );
};

AdditionalPassengerReturnQuery.propTypes = {
  setAdditionalPassengerReturn: PropTypes.func,
};

AdditionalPassengerReturnQuery.defaultProps = {
  setAdditionalPassengerReturn: () => {},
};

export default AdditionalPassengerReturnQuery;
