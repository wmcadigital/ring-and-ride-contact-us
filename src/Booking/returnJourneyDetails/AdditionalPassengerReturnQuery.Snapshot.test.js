import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import AdditionalPassengerReturnQuery from "./AdditionalPassengerReturnQuery";

jest.mock("react-final-form");

describe("AdditionalPassengerReturnQuery - snapshots", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<AdditionalPassengerReturnQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfSomeone",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<AdditionalPassengerReturnQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
        firstName: "John",
      },
      submitFailed: true,
      errors: { additionalPassengerReturn: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<AdditionalPassengerReturnQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
