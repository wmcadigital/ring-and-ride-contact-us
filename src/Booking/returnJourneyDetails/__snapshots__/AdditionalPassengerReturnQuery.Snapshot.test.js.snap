// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalPassengerReturnQuery - snapshots individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their return journey"
    sectionPosition="Section 3 of 4"
  />
  <Question
    text="Does John want to bring anyone else with them on the return journey?"
  />
  <RadioGroup
    error={null}
  >
    <label
      className="wmrards-fe-label"
      htmlFor="additionalPassengerReturn"
    >
      For example, a carer or guardian. They do not need to be a registered Ring and Ride scheme user.
    </label>
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="additionalPassengerReturn"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="additionalPassengerReturn"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`AdditionalPassengerReturnQuery - snapshots individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your return journey"
    sectionPosition="Section 3 of 4"
  />
  <Question
    text="Do you want to bring anyone else with you on the return journey?"
  />
  <RadioGroup
    error={null}
  >
    <label
      className="wmrards-fe-label"
      htmlFor="additionalPassengerReturn"
    >
      For example, a carer or guardian. They do not need to be a registered Ring and Ride scheme user.
    </label>
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="additionalPassengerReturn"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="additionalPassengerReturn"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`AdditionalPassengerReturnQuery - snapshots renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your return journey"
    sectionPosition="Section 3 of 4"
  />
  <Question
    text="Do you want to bring anyone else with you on the return journey?"
  />
  <RadioGroup
    error="Required"
  >
    <label
      className="wmrards-fe-label"
      htmlFor="additionalPassengerReturn"
    >
      For example, a carer or guardian. They do not need to be a registered Ring and Ride scheme user.
    </label>
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="additionalPassengerReturn"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="additionalPassengerReturn"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;
